doc/developer-guides/hld/hv-startup.rst

```diff
@@ -111,70 +112,71 @@ Memory
+After the BSP detects that all APs are up, it continues to enter guest mode.
+Likewise, after one AP completes its initialization, it starts entering guest
+mode as well. When the BSP and APs enter guest mode, they try to launch
+predefined VMs whose vBSP is associated with this physical core. These
+predefined VMs are configured in ``vm config`` and may be a
+pre-launched Safety VM or Service VM.
 
 .. _vm-startup:
 
 VM Startup
 **********
 
 The Service VM or a pre-launched VM is created and launched on the physical
-CPU which configured as its vBSP. Meanwhile, for the physical CPUs which
-configured as vAPs for dedicated VMs, they will enter the default idle loop
+CPU that is configured as its vBSP. Meanwhile, for the physical CPUs that
+are configured as vAPs for dedicated VMs, they enter the default idle loop
 (refer to :ref:`VCPU_lifecycle` for details), waiting for any vCPU to be
 scheduled to them.
 
-:numref:`hvstart-vmflow` illustrates a high-level execution flow of
-creating and launching a VM, applicable to pre-launched VM, Service VM
-and User VM. One major difference in the creation of User VM and pre-launched
-/Service VM is that pre-launched/Service VM is created by the hypervisor,
-while the creation of User VMs is triggered by the DM in Service OS.
-The main steps include:
+:numref:`hvstart-vmflow` illustrates a high-level execution flow of creating and
+launching a VM, applicable to pre-launched User VMs, Service VM, and
+post-launched User VMs. One major difference in the creation of post-launched
+User VMs vs. pre-launched User VMs or Service VM is that the pre-launched User
+VMs and Service VM are created by the hypervisor, while post-launched User VMs
+are created by the Device Model (DM) in the Service VM. The main steps include:
 
 -  **Create VM**: A VM structure is allocated and initialized. A unique
    VM ID is picked, EPT is initialized, e820 table for this VM is prepared,
    I/O bitmap is set up, virtual PIC/IOAPIC/PCI/UART is initialized, EPC for
    virtual SGX is prepared, guest PM IO is set up, IOMMU for PT dev support
    is enabled, virtual CPUID entries are filled, and vCPUs configured in this VM's
-   ``vm config`` are prepared. For post-launched User VM, the EPT page table and
-   e820 table is actually prepared by DM instead of hypervisor.
+   ``vm config`` are prepared. For a post-launched User VM, the EPT page table
+   and e820 table are prepared by the DM instead of the hypervisor.
 
--  **Prepare vCPUs:** Create the vCPUs, assign the physical processor it
-   is pinned to, a unique-per-VM vCPU ID and a globally unique VPID,
-   and initializes its virtual lapic and MTRR, and its vCPU thread object got setup
-   for vcpu scheduling. The vCPU number and affinity are defined in corresponding
+-  **Prepare vCPUs:** Create the vCPUs, assign the physical processor the vCPU
+   is pinned to, a unique-per-VM vCPU ID and a globally unique VPID, initialize
```

Comment:
@dbkinder @fitchbe What does "a unique-per-VM vCPU ID and a globally unique VPID" mean -- seems like it's missing a verb?

Reply:
I'll bet it was meant as a parenthetical commenting on the specifics RE assigning "the physical processor [that] the vCPU is pinned to".